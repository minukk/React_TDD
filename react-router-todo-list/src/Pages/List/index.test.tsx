import React from 'react';
import { Routes, Route, Router, useLocation } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { screen, render, fireEvent } from '@testing-library/react';
import 'jest-styled-components';

import { ToDoListProvider } from 'Context';
import { List } from './index';

describe('<List />', () => {
  it('renders component correctly', () => {
    const history = createMemoryHistory();
    history.push('/');

    localStorage.setItem('ToDoList', '["ToDo 1", "ToDo 2", "ToDo 3"]');

    const { container } = render(
      <ToDoListProvider>
        <Router navigator={history} location={history.location}>
          <List />
        </Router>
      </ToDoListProvider>,
    );

    const toDoItem1 = screen.getByText('ToDo 1');
    expect(toDoItem1).toBeInTheDocument();
    expect(toDoItem1.getAttribute('href')).toBe('/detail/0');
    const toDoItem2 = screen.getByText('ToDo 2');
    expect(toDoItem2).toBeInTheDocument();
    expect(toDoItem2.getAttribute('href')).toBe('/detail/1');
    const toDoItem3 = screen.getByText('ToDo 3');
    expect(toDoItem3).toBeInTheDocument();
    expect(toDoItem3.getAttribute('href')).toBe('/detail/2');

    expect(screen.getAllByText('삭제').length).toBe(3);

    const addButton = screen.getByText('+');
    expect(addButton).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
