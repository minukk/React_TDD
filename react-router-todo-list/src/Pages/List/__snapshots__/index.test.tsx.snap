// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders component correctly 1`] = `
.c4 {
  padding: 10px 20px;
  border-radius: 8px;
  background-color: #ff1744;
  text-align: center;
  cursor: pointer;
}

.c4:hover {
  background-color: #f01440;
}

.c4:active {
  box-shadow: inset 5px 5px 10px rgba(0,0,0,0.2);
}

.c5 {
  color: #fff;
  font-size: 16px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 10px;
  padding: 10px;
  border-bottom: 1px solid #bdbdbd;
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-left: 20px;
  font-size: 16px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  min-width: 350px;
  height: 400px;
  margin-bottom: 20px;
  border: 1px solid #bdbdbd;
  overflow-y: scroll;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 20px;
  border-radius: 8px;
  background-color: #fff;
  box-shadow: 5px 5px 10px rgba(0,0,0,0.2);
}

.c6 {
  position: absolute;
  bottom: -30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 60px;
  height: 60px;
  border-radius: 30px;
  background-color: #304ffe;
  font-size: 20px;
  color: #fff;
  box-shadow: 5px 5px 10px rgba(0,0,0,0.2);
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c6:hover {
  background-color: #1e40ff;
}

.c6:active {
  box-shadow: inset 5px 5px 10px rgba(0,0,0,0.2);
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      data-testid="toDoList"
    >
      <div
        class="c2"
      >
        <a
          class="c3"
          href="/detail/0"
        >
          ToDo 1
        </a>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            삭제
          </div>
        </div>
      </div>
      <div
        class="c2"
      >
        <a
          class="c3"
          href="/detail/1"
        >
          ToDo 2
        </a>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            삭제
          </div>
        </div>
      </div>
      <div
        class="c2"
      >
        <a
          class="c3"
          href="/detail/2"
        >
          ToDo 3
        </a>
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            삭제
          </div>
        </div>
      </div>
    </div>
    <a
      class="c6"
      href="/add"
    >
      +
    </a>
  </div>
</div>
`;
